### PR TITLE
Edit release notes for Mimir 2.3

### PR DESCRIPTION
**Pull request by pianohacker**
#### What this PR does

Edits the release notes for Mimir 2.3.

### CODE REVIEW

**Comment by replay** [comment]
FYI the CI is failing.

One error seems to be because you need to do `make doc`, I don't know what the other one is about, let's see if `make doc` fixes it.

File: docs/sources/release-notes/v2.3.md [hunk 1]
```diff
@@ -9,53 +9,57 @@ weight: 300
 
 Grafana Labs is excited to announce version 2.3 of Grafana Mimir, the most scalable, most performant open source time series database in the world.
 
-The highlights that follow include the top features, enhancements, and bugfixes in this release. If you are upgrading from Grafana Mimir 2.2, there is upgrade-related information as well.
-For the complete list of changes, see the [Changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
+The highlights that follow include the top features, enhancements, and bugfixes in this release. For the complete list of changes, see the [Changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.3.md:12]: See https://github.com/grafana/technical-documentation/blob/b685d35af697bd905f340658ed06e38fd6d8ab52/docs/sources/style-guide/word-list.md.
```suggestion
The highlights that follow include the top features, enhancements, and bugfixes in this release. For the complete list of changes, see the [changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
```

File: docs/sources/release-notes/v2.3.md [hunk 2]
```diff
@@ -9,53 +9,57 @@ weight: 300
...
+
+**Note**: If you are upgrading from Grafana Mimir 2.2, review the [list of important changes](#important-changes) below.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.3.md:14]:
```suggestion
**Note**: If you are upgrading from Grafana Mimir 2.2, review the [list of important changes](#important-changes) that follow.
```
Rationale: avoid directional writing in tech documentation

File: docs/sources/release-notes/v2.3.md [hunk 3]
```diff
@@ -9,53 +9,57 @@ weight: 300
...
+- [Query sharding](https://grafana.com/docs/mimir/latest/operators-guide/architecture/query-sharding/) is now enabled by default which should give you better performance on high cardinality metrics queries.
+  - To compensate for the increased number of queries generated by query sharding, the [query scheduler](https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/query-scheduler/) component is now enabled by default.
+- The backfill API endpoints for importing historic time series data are now exposed on the Nginx gateway.
+- Nginx now sets the value of the `X-Scope-OrgID` header equal to the value of Mimir's `no_auth_tenant` parameter by default. The previous release had set the value of `X-Scope-OrgID` to `anonymous` by default which [complicated the process of migrating to Mimir](https://github.com/grafana/mimir/issues/2563).
+- Memberlist now uses DNS service-discovery by default, which should decrease startup time for large Mimir clusters.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.3.md:47]:
```suggestion
- Memberlist now uses DNS service-discovery by default, which decreases startup time for large Mimir clusters.
```
Should is ambiguous. Will it or will it not?

File: docs/sources/release-notes/v2.3.md [hunk 4]
```diff
@@ -9,53 +9,57 @@ weight: 300
...
+
+Also, as [noted above](#helm-chart-improvements), the administrator password for Helm chart deployments using the built-in MinIO is now set differently.
 
 ## Bug fixes
 
-- [PR 2447](https://github.com/grafana/mimir/pull/2447): Fix incorrect mapping of http status codes `429` to `500` when the request queue is full in the query-frontend. This corrects behavior in the query-frontend where a `429 "Too Many Outstanding Requests"` error (a retriable error) from a querier was incorrectly returned as a `500` system error (an unretriable error).
+- [PR 2447](https://github.com/grafana/mimir/pull/2447): Fix incorrect mapping of http status codes `429` to `500` when the request queue is full in the query-frontend. This corrects behavior in the query-frontend where a `429 "Too Many Outstanding Requests"` error (a retryable error) from a querier was incorrectly returned as a `500` system error (an unretriable error).
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.3.md:63]:
```suggestion
- [PR 2447](https://github.com/grafana/mimir/pull/2447): Fix incorrect mapping of http status codes `429` to `500` when the request queue is full in the query-frontend. This corrects behavior in the query-frontend where a (retryable) `429 "Too Many Outstanding Requests"` error from a querier was incorrectly returned as a `500` system error (an unretriable error).
```

Comment (osg-grafana) [reply docs/sources/release-notes/v2.3.md:63]: Also fine as-is; you decide. :)

**Review by osg-grafana — APPROVED** [review]
Some minor feedback included in review

**Review by krajorama — APPROVED** [review]

**Pull request merged**